### DISCUSSION TITLE
Bill all subscriptions at a set date of each month?

### DISCUSSION THREAD

**Discussion by shc023** [swellstores/community#544] [Help] [answered]
We have a subscription product that we fulfill in bulk `on the 1st of each month`, whoever joins the program in the middle of the month will just get their first fulfillment on the next cycle (e.g. 1st of the next month).

Is it possible to set something on the subscription product such that its billed and creates the order on a specific day of the month? or alternatively, is there any way to programmatically update an existing subscription's next charge date to achieve the same effect after the fact?

**Comment by sags95** [comment, marked as answer]
It's not possible to specify a calendar day for billing yet, but it would be possible to programatically update a subscription. 

For a new subscription, is the idea to bill on checkout and only create an order on the 1st of the month, or create the subscription on checkout and only charge/create order on the 1st? The simplest way would be to dynamically update the `trial_days` on the cart item to the 1st of the month. With the approach the customer can checkout, enter billing info, and the subscription is created but not charged until the 1st of the month (end of trial).

**Reply by shc023** [reply]
@sags95 that might be an acceptable workaround - if I programmatically calculate `trial_days` to number of days until the 1st next month. Is `SwellCartItemPurchaseOption.trialDays` the right model prop to be looking at?

For existing subscriptions, what property should I be looking to update to set the next billing date? `datePeriodStart`?
